### PR TITLE
[plugin-rest-api] Align syntax of REST api steps

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 1]
```diff
@@ -421,6 +427,192 @@ When I execute HTTP $httpMethod request for resource with URL `$url`
...
+
+.Execute HTTP GET request
+[source,gherkin]
+----
+When I execute HTTP GET request for resource with relative URL `/image/jpeg`
+----
+
+=== Compare response body with resource data
+
+Compares the response body data with the resource data accorging to provided rule.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:462]: the same as for title

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 2]
```diff
@@ -421,6 +427,192 @@ When I execute HTTP $httpMethod request for resource with URL `$url`
...
+
+[source,gherkin]
+----
+Then response code is $comparisonRule `$responseCode`
+----
+
+* `$comparisonRule` - xref:parameters:comparison-rule.adoc[The comparison rule].
+* `$responseCode` - The expected {http-status-code}.
+
+.Resource is not found
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:491]:
```suggestion
.Validate the resource is not found
```

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 3]
```diff
@@ -421,6 +427,192 @@ When I execute HTTP $httpMethod request for resource with URL `$url`
...
+[source,gherkin]
+----
+When I execute HTTP GET request for resource with URL `https://httpbin.org/cfa2fdbc`
+Then response code is equal to `404`
+----
+
+=== Validate response time
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:498]: `response` -> `HTTP response` - everywhere

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 4]
```diff
@@ -421,6 +427,192 @@ When I execute HTTP $httpMethod request for resource with URL `$url`
...
+.Check response time
+[source,gherkin]
+----
+When I execute HTTP GET request for resource with URL `https://example.com`
+Then response time is less than `1000` milliseconds
+----
+
+=== Valdate number of response headers
+
+Check the number of specified by name response headers.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:519]: `Check` -> `Validates`

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 5]
```diff
@@ -879,3 +1087,40 @@ Then HTTP resources are valid:
 |https://saucelabs.com                                  |
 |https://vividus-test-site.herokuapp.com/img/vividus.png|
 ----
+
+== Tips
+
+.Validate response body
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:1093]: it makes sense to add some title and description for examples

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 6]
```diff
@@ -879,3 +1087,40 @@ Then HTTP resources are valid:
 |https://saucelabs.com                                  |
 |https://vividus-test-site.herokuapp.com/img/vividus.png|
 ----
+
+== Tips
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:1091]:
```suggestion
== Tips & Tricks
```

File: vividus-plugin-rest-api/src/main/java/org/vividus/steps/api/HttpResponseValidationSteps.java [hunk 7]
```diff
@@ -71,17 +75,25 @@ public HttpResponseValidationSteps(HttpTestContext httpTestContext, VariableCont
...
+    {
+        LOGGER.warn("The step: \"Then the response does not contain body\" is deprecated and will be removed in VIVIDUS"
+                + " 0.6.0. Use step: \"Then response does not contain body\"");
+        doesResponseContainNoBody();
+    }
+
+    /**
+     * Checks that the HTTP response does not contain a body
+     *
+     * <p><i>Step can't be used without previous step with HTTP-method</i></p>
```
Comment (valfirst) [inline vividus-plugin-rest-api/src/main/java/org/vividus/steps/api/HttpResponseValidationSteps.java:93]: it requires clarification

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — COMMENTED** [review]
is a file with deprecated steps lost?

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 8]
```diff
@@ -161,7 +161,9 @@ Then `${response}` matches `.+Home.+`
 
 === Response code
 
-Provides the https://developer.mozilla.org/en-US/docs/Web/HTTP/Status[HTTP response status code].
+:http-status-code: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status[HTTP response status code]
+
+Provides the {http-status-code}.
 
 ==== *Variable name*
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:168]:
```suggestion
```

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 9]
```diff
@@ -421,6 +423,192 @@ When I execute HTTP $httpMethod request for resource with URL `$url`
 When I execute HTTP GET request for resource with URL `https://example.com`
 ----
 
+=== Execute HTTP request with relative URL
+
+Executes the HTTP request for a resource identified by the URL relative to the base URL specified by the `rest-api.http.endpoint` property.
+
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:429]: do not put each sentence as a separate paragraph

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 10]
```diff
@@ -421,6 +423,192 @@ When I execute HTTP $httpMethod request for resource with URL `$url`
...
+.Compare images
+[source,gherkin]
+----
+When I execute HTTP GET request for resource with URL `https://httpbin.org/image/png`
+Then response body is equal to resource at `/images/pig.png`
+----
+
+=== Validate HTTP response code
+
+Compares the {http-status-code} with the expected number.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:475]:
```suggestion
Compares the {http-status-code} against the expected number.
```

Comment (valfirst) [reply docs/modules/plugins/pages/plugin-rest-api.adoc:475]: here and everywhere else

**Review by valfirst — CHANGES_REQUESTED** [review]

File: docs/modules/plugins/pages/plugin-rest-api.adoc (hunk 9 above)
Comment (uarlouski) [reply docs/modules/plugins/pages/plugin-rest-api.adoc:429]: the sentences are completely different by their meaning, so why not to separate them logically into paragraphs?

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 11]
```diff
@@ -395,13 +397,11 @@ When I send HTTP GET to the relative URL '/get?name=Content'
...
-=== Execute HTTP request
-
-Executes the HTTP request to access a resource on the server identified by the URL.
+=== Execute HTTP request with absolute URL
 
-The step uses previously set HTTP headers and request body.
-
-The response HTTP headers, response status code and response body can be accessed by the corresponding steps and dynamic variables.
+Executes the HTTP request for a resource identified by the absolute URL. The step uses previously set HTTP headers
+and request body. The HTTP response headers, status code and body of the latest executed HTTP call can be accessed
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:403]:
```suggestion
and request body. The HTTP response headers, status code and body can be accessed
```

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 12]
```diff
@@ -395,13 +397,11 @@ When I send HTTP GET to the relative URL '/get?name=Content'
...
+by the corresponding steps and dynamic variables.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:404]:
```suggestion
via the corresponding steps and dynamic variables.
```

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 13]
```diff
@@ -417,6 +417,190 @@ When I execute HTTP $httpMethod request for resource with URL `$url`
 When I execute HTTP GET request for resource with URL `https://example.com`
 ----
 
+=== Execute HTTP request with relative URL
+
+Executes the HTTP request for a resource identified by the URL relative to the base URL specified by the `rest-api.http.endpoint`
+property. The step uses previously set HTTP headers and request body. The HTTP response headers, status code and body of the latest
+executed HTTP call can be accessed by the corresponding steps and dynamic variables.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:424]:
```suggestion
property. The step uses previously set HTTP headers and request body. The HTTP response headers, status code and body can be accessed via the corresponding steps and dynamic variables.
```

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 14]
```diff
@@ -417,6 +417,190 @@ When I execute HTTP $httpMethod request for resource with URL `$url`
...
+----
+When I execute HTTP GET request for resource with URL `https://example.com`
+Then response time is less than `1000` milliseconds
+----
+
+=== Valdate number of HTTP response headers
+
+:http-response-header: https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Response_fields[HTTP response header]
+
+Validates the number of specified by name HTTP response headers.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:507]:
```suggestion
Validates the number of HTTP response headers filtered by the specified name.
```

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 15]
```diff
@@ -417,6 +417,190 @@ When I execute HTTP $httpMethod request for resource with URL `$url`
...
+.Check HTTP response content type
+[source,gherkin]
+----
+When I execute HTTP GET request for resource with URL `https://httpbin.org/robots.txt`
+Then value of response header `Content-Type` is equal to `text/plain`
+----
+
+=== Validate HTTP response header attributes
+
+Validates that the response header with the specified name contains attributes.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:546]: it's needed to describe what attributes are

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 16]
```diff
@@ -861,3 +1061,50 @@ The step xref:plugins:plugin-json.adoc#_save_number_of_json_elements[is moved] t
...
+=== Validate HTTP response body
+
+In order to validate the body of the latest HTTP response use the <<_response>> dynamic variable to access the body
+data in conjunction with any step responsible for data validation like data comparison, JSON, XML and other steps.
+
+.Validate response body
+[source,gherkin]
+----
+When I execute HTTP GET request for resource with URL `https://httpbin.org/json`
+Then `${response}` is equal to `
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:1076]: for JSON comparison it's better to use JSON-specific steps

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 17]
```diff
@@ -861,3 +1061,50 @@ The step xref:plugins:plugin-json.adoc#_save_number_of_json_elements[is moved] t
...
+      }
+    ], 
+    "title": "Sample Slide Show"
+  }
+}`
+----
+
+=== Save HTTP response body into variable
+
+The HTTP response body can be saved into variable by using the xref:commons:vividus-steps.adoc#_initialize_variable[init variable step]
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:1102]:
```suggestion
The HTTP response body can be saved into static variable by using the xref:commons:vividus-steps.adoc#_initialize_variable[init variable step]
```

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 18]
```diff
@@ -861,3 +1061,50 @@ The step xref:plugins:plugin-json.adoc#_save_number_of_json_elements[is moved] t
...
+in conjunction with the <<_response>> dynamic variable.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:1103]: please describe an example when this may be needed

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Pull request merged**